### DISCUSSION TITLE
Add project publishing (Discuss storage options)

### DISCUSSION THREAD

**Discussion by icarusinthewings** [NovaoTech/solaris#3] [Development]
Possible Systems:
IPFS (Cheapest)
Google Cloud (Less Control, costs more money)
Self-hosted by J Technologies (Complete Control, costs less, uses more storage as grows. Possible downtime)

**Comment by icarusinthewings** [comment]
So far I think IPFS would be best for the scale we are at.

**Reply by icarusinthewings** [reply]
@micahlt  Hi lol

**Reply by icarusinthewings** [reply]
> @micahlt Hi lol

Kinda hoping you guys could help with this

**Reply by icarusinthewings** [reply]
Seeing that the ST has ruined scratch even more somehow

**Reply by micahlt** [reply]
@VelocityDesign Just want to say that this is an enormous undertaking and you may not know what you're getting yourself into.  You need hosting, and servers, and a framework choice, and all kinds of other stuff.

**Reply by icarusinthewings** [reply]
@micahlt I have a new 8g ram server I could host this on. So far I am looking into using Flask as my framework.

**Reply by micahlt** [reply]
@VelocityDesign Flask?  Aight, but I'll warn you I know very little about Python.

**Reply by icarusinthewings** [reply]
@micahlt That's fine. Flask uses HTML for templates so it should be easier to help. Flask loads the dynamic content and inserts it into the HTML.

**Reply by micahlt** [reply]
Sounds cool 😄 I still can't commit to give a certain amount of time to this, though.  I've got a lot of other stuff on my plate.  Before working out the nitty-gritty details, I'd highly recommend you write up a document detailing how this project is going to be governed, what the goal is, and why it exists.  That way everything is easy to understand.

**Reply by MikeDev101** [reply]
* A Python programmer approaches!

I'm not familiar with Flask, so I'll be taking a look into it. I also agree with @micahlt that we should have an outline for the project. Also, I'd personally pick a self-hosted solution, but that's mainly because I'm not familiar with IPFS and Google's policies/services change quite often(?)

**Reply by icarusinthewings** [reply]
@MikeDev101  Welcome! So far, I've implemented IPFS into the code.  I think IPFS is the better choice for now because it would allow us to store more projects. I'll just be using Pinata API to pin projects and Dweb.link to get them.

**Reply by icarusinthewings** [reply]
However, self hosted could work possibly.

**Reply by MikeDev101** [reply]
> @MikeDev101 Welcome! So far, I've implemented IPFS into the code. I think IPFS is the better choice for now because it would allow us to store more projects. I'll just be using Pinata API to pin projects and Dweb.link to get them.

Any updates?

**Reply by icarusinthewings** [reply]
> > @MikeDev101 Welcome! So far, I've implemented IPFS into the code. I think IPFS is the better choice for now because it would allow us to store more projects. I'll just be using Pinata API to pin projects and Dweb.link to get them.
> 
> Any updates?

I'm mostly trying to figure out how to implement saving into the Turbowarp Editor.

**Reply by CloudGate1234** [reply]
> > > @MikeDev101 Welcome! So far, I've implemented IPFS into the code. I think IPFS is the better choice for now because it would allow us to store more projects. I'll just be using Pinata API to pin projects and Dweb.link to get them.
> > 
> > 
> > Any updates?
> 
> I'm mostly trying to figure out how to implement saving into the Turbowarp Editor.

hello. speaking of IPFS, you should get a "unstoppable domain" I recommend polarium.zil ; only 20$ and no renewal fees FOREVER. unstoppable domains about: https://medium.com/unstoppabledomains/intro-to-blockchain-domain-nfts-bba976942272

**Reply by icarusinthewings** [reply]
> > > > @MikeDev101 Welcome! So far, I've implemented IPFS into the code. I think IPFS is the better choice for now because it would allow us to store more projects. I'll just be using Pinata API to pin projects and Dweb.link to get them.
> > > 
> > > 
> > > Any updates?
> > 
> > 
> > I'm mostly trying to figure out how to implement saving into the Turbowarp Editor.
> 
> hello. speaking of IPFS, you should get a "unstoppable domain" I recommend polarium.zil ; only 20$ and no renewal fees FOREVER. unstoppable domains about: https://medium.com/unstoppabledomains/intro-to-blockchain-domain-nfts-bba976942272

I have an unstoppable domain for J Tech :)
